Unable to save plain EmbededMap

The following statements against orientDB 2.0.2 fails:
```
db.insert().into("TestEmbededMap").set({
                name: "abc",
                map: {
                    "k1": "v1",
                    "k2": "v2"
                }
            }).one();
```

TestEmbededMap is a simple class with an plain EmbededMap field.

The server returns:
```
OrientDB.RequestError: Error on execution of command: sql.INSERT INTO TestEmbededMap SET name = "abc", map = {"@type":"d","k1":"v1","k2":"v2"}
```

and the server logs:
```
com.orientechnologies.orient.core.exception.OCommandExecutionException: Error on execution of command: sql.INSERT INTO TestEmbededMap SET name = "abc", map = {"@type":"d","k1":"v1","k2":"v2"}
	at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.executeCommand(OAbstractPaginatedStorage.java:1190)
	at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.command(OAbstractPaginatedStorage.java:1173)
	at com.orientechnologies.orient.core.command.OCommandRequestTextAbstract.execute(OCommandRequestTextAbstract.java:63)
	at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.command(ONetworkProtocolBinary.java:1178)
	at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.executeRequest(ONetworkProtocolBinary.java:385)
	at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.execute(OBinaryNetworkProtocolAbstract.java:216)
	at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:65)
Caused by: java.lang.ClassCastException: com.orientechnologies.orient.core.record.impl.ODocument cannot be cast to java.util.Map
	at com.orientechnologies.orient.core.record.impl.ODocument.autoConvertValues(ODocument.java:2018)
	at com.orientechnologies.orient.core.record.impl.ODocument.validate(ODocument.java:1898)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.save(ODatabaseDocumentTx.java:2218)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.save(ODatabaseDocumentTx.java:117)
	at com.orientechnologies.orient.core.record.impl.ODocument.save(ODocument.java:1704)
	at com.orientechnologies.orient.core.record.impl.ODocument.save(ODocument.java:1695)
	at com.orientechnologies.orient.core.record.impl.ODocument.save(ODocument.java:75)
	at com.orientechnologies.orient.core.sql.OCommandExecutorSQLInsert.saveRecord(OCommandExecutorSQLInsert.java:306)
	at com.orientechnologies.orient.core.sql.OCommandExecutorSQLInsert.execute(OCommandExecutorSQLInsert.java:207)
	at com.orientechnologies.orient.core.sql.OCommandExecutorSQLDelegate.execute(OCommandExecutorSQLDelegate.java:64)
	at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.executeCommand(OAbstractPaginatedStorage.java:1184)
	... 6 more
```

seems it could succeed without *"@type" : "d"*

--- a/lib/utils.js
+++ b/lib/utils.js
@@ -214,20 +214,12 @@ exports.encode = function encode (value) {
   else {
 
     var keys = Object.keys(value),
-        offset = 0,
         length = keys.length;
 
-    if (length && !value['@type']) {
-      offset = 1;
-    }
-
-    var parts = new Array(length + offset),
+    var parts = new Array(length),
         key, i;
-    if (offset) {
-      parts[0] = '"@type":"d"';
-    }
-    for (i = offset; i < length + offset; i++) {
-      key = keys[i - offset];
+    for (i = 0; i < length; i++) {
+      key = keys[i];
       parts[i] = '"' + exports.escape(key) + '":'+encode(value[key]);
     }
     return '{'+parts.join(',')+'}';

--- a/test/bugs/252-save-embedded-map.js
+++ b/test/bugs/252-save-embedded-map.js
@@ -1,0 +1,68 @@
+describe("Bug #252: Unable to save plain EmbededMap", function () {
+  before(function () {
+    return CREATE_TEST_DB(this, 'testdb_bug_252')
+    .bind(this)
+    .then(function () {
+      return this.db.class.create('TestEmbeddedMap');
+    })
+    .then(function (TestEmbeddedMap) {
+      return TestEmbeddedMap.property.create([
+        {
+          name: 'name',
+          type: 'string'
+        },
+        {
+          name: 'map',
+          type: 'embeddedmap',
+          linkedType: 'string'
+        },
+        {
+          name: 'list',
+          type: 'embeddedlist'
+        }
+      ])
+    });
+  });
+  after(function () {
+    return DELETE_TEST_DB('testdb_bug_252');
+  });
+
+  it('should insert a map into the database', function () {
+    return this.db
+    .insert()
+    .into('TestEmbeddedMap')
+    .set({
+      name: 'abc',
+      map: {
+        k1: 'v1',
+        k2: 'v2'
+      }
+    })
+    .one()
+    .then(function (res) {
+      res.map.k1.should.equal('v1');
+    });
+  });
+
+  describe('Bug #255: param is not working with embeddedlist', function () {
+    it('should allow params in embedded list', function () {
+      return this.db.query('INSERT INTO TestEmbeddedMap SET name = :name, list = :list', {
+        params: {
+          name: 'def',
+          list: [
+            {
+              controller: 'home'
+            }
+          ]
+        }
+      })
+      .bind(this)
+      .then(function () {
+        return this.db.select().from('TestEmbeddedMap').where({name: 'def'}).one();
+      })
+      .then(function (row) {
+        row.list[0].controller.should.equal('home');
+      });
+    });
+  });
+});

